### ISSUE TITLE
NFS Fails to work on OCP 4.4.3 - The Deployment "nfs-client-provisioner" is invalid

### ISSUE THREAD

**Issue by cwawak** [RedHatOfficial/ocp4-helpernode#51]
When running `helpernodecheck nfs-setup`, I didn't necessarily remember seeing errors, but I couldn't get PVCs to create a new PV.

I did some investigating, and ran these commands:

```
[root@helper src]# oc project nfs-provisioner
Already on project "nfs-provisioner" on server "https://api.ocp4.melrose.wawak.org:6443".
[root@helper src]# oc version
Client Version: openshift-clients-4.3.0-201910250623-88-g6a937dfe
Server Version: 4.4.3
Kubernetes Version: v1.17.1
[root@helper src]# oc create -f nfs-provisioner-deployment.yaml
The Deployment "nfs-client-provisioner" is invalid:
* spec.selector: Required value
* spec.template.metadata.labels: Invalid value: map[string]string{"app":"nfs-client-provisioner"}: `selector` does not match template `labels`
[root@helper src]# oc project
Using project "nfs-provisioner" on server "https://api.ocp4.melrose.wawak.org:6443".
[root@helper src]# oc get all
No resources found in nfs-provisioner namespace.
```

I'm not sure exactly what's going on, but there seems to be some sort of incompatibility between 4.4.3 and the nfs-provisoner stuff.

Do you need a must gather? This environment worked perfectly with the nfs-client and 4.3.

**Comment by christianh814** [comment]
The issue was a deprecated API section. Updated the deployment in the `devel` branch 44d30ae3ce0e741d6b36e7378886f744ad94e32c

Will keep this open until it merges with `master`.

**Comment by cwawak** [comment]
I moved to the devel branch, and the nfs-provisioner works again! I still had to approve CSRs though.

`# /usr/local/bin/helpernodecheck nfs-setup
Now using project "nfs-provisioner" on server "https://api.ocp4.melrose.wawak.org:6443".

You can add applications to this project with the 'new-app' command. For example, try:

    oc new-app ruby~https://github.com/sclorg/ruby-ex.git

to build a new example application in Python. Or use kubectl to deploy a simple Kubernetes application:

    kubectl create deployment hello-node --image=gcr.io/hello-minikube-zero-install/hello-node

Already on project "nfs-provisioner" on server "https://api.ocp4.melrose.wawak.org:6443".
serviceaccount/nfs-client-provisioner created
clusterrole.rbac.authorization.k8s.io/nfs-client-provisioner-runner created
clusterrolebinding.rbac.authorization.k8s.io/run-nfs-client-provisioner created
role.rbac.authorization.k8s.io/leader-locking-nfs-client-provisioner created
rolebinding.rbac.authorization.k8s.io/leader-locking-nfs-client-provisioner created
securitycontextconstraints.security.openshift.io/hostmount-anyuid added to: ["system:serviceaccount:nfs-provisioner:nfs-client-provisioner"]
deployment.apps/nfs-client-provisioner created
storageclass.storage.k8s.io/nfs-storage-provisioner created
storageclass.storage.k8s.io/nfs-storage-provisioner annotated
Now using project "default" on server "https://api.ocp4.melrose.wawak.org:6443".
Waiting for deployment "nfs-client-provisioner" rollout to finish: 0 of 1 updated replicas are available...
deployment "nfs-client-provisioner" successfully rolled out

Deployment started; you should monitor it with "oc get pods -n nfs-provisioner"

[root@helper ~]# oc get pods -n nfs-provisioner
NAME                                      READY   STATUS    RESTARTS   AGE
nfs-client-provisioner-69c45db645-hwdgj   1/1     Running   0          14s

`

**Comment by christianh814** [comment]
>  I still had to approve CSRs though. 

Yes, and I can't quite figure out why. On 4.3 it was "automagically" Still investigating

**Comment by christianh814** [comment]
Merged change to master. Closed by 8a01856cd39b5c54861c1050391af29bfd76922b
